### PR TITLE
Refactor /components/tf_ng_tensorboard to /webapp

### PR DESCRIPTION
**Pull request by davidsoergel**
This refactors `/components/tf_ng_tensorboard` into a new `/webapp` directory, structured as agreed per internal doc.  Aside from file moves and associated import updates, some changes are:

* Consolidate build rules into `tensorboard/webapp/BUILD` (including two from `/tensorboard/BUILD`).
* Reorder rules in `tensorboard/webapp/BUILD` according to their dependency structure.  The main app component is wrapped in 6 layers of rules- all for good reason, but I was too easily confused when they were out of order.
* Moved `core.service.ts` to `webapp/client`, with its own `ClientService` module.  I know we planned to avoid Services but I now think we should reconsider that.

```
tensorboard/webapp
├── angular
│   └── BUILD
├── app.component.css
├── app.component.html
├── app.component.ts
├── app.module.ts
├── bootstrap.ts
├── BUILD
├── client
│   ├── BUILD
│   ├── client.module.ts
│   └── client.service.ts
├── config.dev.ts
├── config.prod.ts
├── core
│   ├── actions
│   │   ├── core.actions.ts
│   │   └── index.ts
│   ├── BUILD
│   ├── core.module.ts
│   ├── effects
│   │   ├── core.effects.test.ts
│   │   ├── core.effects.ts
│   │   └── index.ts
│   ├── store
│   │   ├── core.reducers.test.ts
│   │   ├── core.reducers.ts
│   │   ├── core.selectors.ts
│   │   ├── core.types.ts
│   │   └── index.ts
│   └── testing
│       ├── BUILD
│       └── index.ts
├── header
│   ├── BUILD
│   ├── header.component.css
│   ├── header.component.html
│   ├── header.component.ts
│   ├── header.container.ts
│   ├── header.module.ts
│   ├── header.test.ts
│   └── types.ts
├── index.html
├── main.prod.ts
├── main.ts
├── plugins
│   ├── BUILD
│   ├── plugins.container.test.ts
│   ├── plugins.container.ts
│   └── plugins.module.ts
├── reducers.ts
├── reloader
│   ├── BUILD
│   ├── reloader.component.test.ts
│   ├── reloader.component.ts
│   └── reloader.module.ts
├── settings
│   ├── BUILD
│   ├── dialog.component.css
│   ├── dialog.component.test.ts
│   ├── dialog.component.ts
│   ├── settings_button.component.ts
│   └── settings.module.ts
├── testing
│   ├── BUILD
│   ├── initialize_testbed.ts
│   └── rxjs_shims.js
└── types
    ├── api.ts
    └── BUILD
```

### CODE REVIEW

**Comment by davidsoergel** [comment]
BTW this incorporates #2886, on the theory that one will be submitted first anyway.  Of course I'll propagate any changes.

File: tensorboard/webapp/core/BUILD [hunk 1]
```diff
@@ -7,15 +7,20 @@ load("//tensorboard/defs:defs.bzl", "tf_ts_library")
...
-        "core.actions.ts",
-        "core.effects.ts",
+        "actions/core.actions.ts",
+        "actions/index.ts",
+        "effects/core.effects.ts",
+        "effects/index.ts",
         "core.module.ts",
-        "core.reducers.ts",
-        "core.service.ts",
+        "store/core.reducers.ts",
```
Comment (stephanwlee) [inline tensorboard/webapp/core/BUILD]: Consider having BUILD file per directory so they can be depended on without the UI parts. Sooner or later, we will, for testing (will be limited to `testing.ts` mostly), have dependency dag pointing across folder.

File: tensorboard/webapp/core/effects/core.effects.test.ts [hunk 2]
```diff
@@ -23,19 +24,18 @@ import {MockStore, provideMockStore} from '@ngrx/store/testing';
...
-import {PluginsListing, LoadState as DataLoadState} from '../types/api';
+import {PluginsListing, LoadState as DataLoadState} from '../../types/api';
+import {ClientService} from '../../client/client.service';
 
 describe('core.effects', () => {
   let httpMock: HttpTestingController;
   let coreEffects: CoreEffects;
   let action: ReplaySubject<Action>;
-  let store: MockStore<State>;
+  let store: MockStore<{}>;
```
Comment (stephanwlee) [inline tensorboard/webapp/core/effects/core.effects.test.ts]: Why the code change to this move PR?

File: tensorboard/webapp/core/actions/index.ts [hunk 3]
```diff
@@ -12,10 +12,5 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
 limitations under the License.
 ==============================================================================*/
-.plugins {
-  height: 100%;
-}
 
-iframe {
-  border: 0;
-}
+export * from './core.actions';
```
[the remark this answers is not in the document]
Comment (davidsoergel) [reply tensorboard/webapp/core/actions/index.ts]: Huh, this is obviously borked, and I'm pretty sure it didn't show up this way in the PR-creation diff.  Anyhow yes let me do some git juggling to clear this up (probably via an upstream move PR, which I agree I should have done in the first place).

**Comment by davidsoergel** [comment]
Abandoning this one in favor of a new chain of broken-up PRs, starting with #2891 .

(I'll address the comments here in the new PRs, of course)

**Pull request closed without merging**